grr/server/grr_response_server/fleetspeak_utils.py

```diff
@@ -112,3 +113,22 @@ def GetLabelsFromFleetspeak(client_id):
       grr_labels.append(fs_label.label)
 
   return grr_labels
+
+
+def FetchClientResourceUsageRecordsFromFleetspeak(
+    client_id: Text, limit: int) -> List[ClientResourceUsageRecord]:
+  """Returns aggregated resource usage metrics of a client in Fleetspeak-enabled database.
+
+  Args:
+    client_id: Id of the client to fetch Fleetspeak resource usage records for.
+    limit: Max number of resource usage records to retrieve.
+
+  Returns:
+    A list of client resource usage records retrieved from Fleetspeak.
+  """
+  res = fleetspeak_connector.CONN.outgoing.FetchClientResourceUsageRecords(
+      admin_pb2.FetchClientResourceUsageRecordsRequest(
+          client_id=GRRIDToFleetspeakID(client_id), limit=limit))
+  if not res or not res.records or not res.records[0]:
```

Comment:
In what edge case does the response have at least one record, but the record would evaluate to false? Please clarify in a comment in the code and add a test case to trigger this edge case.

Reply:
You're right, I forgot that empty lists in Python will [evaluate as false](https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty) :)